getting_started/scripting/cross_language_scripting.rst

```diff
@@ -0,0 +1,213 @@
+
+Accessing fields
+----------------
+
+Accessing C# fields from GDScript
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Accessing C# fields from GDScript is straightforward, you shouldn't have
+anything to worry about.
+
+.. code-block:: gdscript
+
+    print(my_csharp_node.str1) # bar
+    my_csharp_node.str1 = "BAR"
+    print(my_csharp_node.str1) # BAR
+
+    print(my_csharp_node.str2) # barbar
+    # my_csharp_node.str2 = "BARBAR" # This line will hang and crash
+
+Note that it doesn't matter if the field is defined as a property or an
+attribute, but trying to set a value on a property that does not define a
+setter will result in a crash.
+
+Accessing GDScript fields from C#
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As C# is statically typed, accessing GDScript from C# is a bit more
+convoluted, you will have to use :ref:`Object.Get() <class_Object_method_get>`
+and :ref:`Object.Set() <class_Object_method_set>`. The first argument is the name of the field you want to access.
+
+.. code-block:: csharp
+
+    GD.Print(myGDScriptNode.Get("str1")); // foo
+    myGDScriptNode.Set("str1", "FOO");
+    GD.Print(myGDScriptNode.Get("str1")); // FOO
+
+    GD.Print(myGDScriptNode.Get("str2")); // foofoo
+    // myGDScriptNode.Set("str2", "FOOFOO"); // This line won't do anything
+
+Keep in mind that when setting a field value you should only use types the
+GDScript side knows about.
+Essentially, you want to work with built-in types as described in :ref:`doc_gdscript` or classes extending :ref:`class_Object`.
+
+Calling methods
+---------------
+
+Calling C# methods from GDScript
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Again, calling C# methods from GDScript should be straightforward. The
+marshalling process will do its best to cast your the arguments to match
+function signatures.
+If that's impossible you'll see the following error: ``Invalid call. Nonexistent function `FunctionName```.
+
+.. code-block:: gdscript
+
+    my_csharp_node.PrintNodeName(self) # myGDScriptNode
+    # my_csharp_node.PrintNodeName() # This line will fail.
+
+    my_csharp_node.PrintArray(["a", "b", "c"]) # a, b, c
+    my_csharp_node.PrintArray([1, 2, 3]) # 1, 2, 3
+
+Calling GDScript methods from C#
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To call GDScript methods from C# you'll need to use
+:ref:`Object.Call() <class_Object_method_call>`. The first arguments is the
+name of the method you want to call. The following arguments will be passed
+to said method.
+
+.. code-block:: csharp
+
+    myGDScriptNode.Call("print_node_name", this); // my_csharp_node
+    // myGDScriptNode.Call("print_node_name"); // This line will fail silently and won't error out.
+
+    String[] arr = new String[] { "a", "b", "c" };
+    myGDScriptNode.Call("print_array", arr); // This line will fail silently and won't error out.
+    myGDScriptNode.Call("print_array", new object[] { arr }); // a, b, c
+    myGDScriptNode.Call("print_array", new object[] { new int[] { 1, 2, 3 } }); // 1, 2, 3
```

Comment:
This needs to be made much more clear. It's too confusing like this.

`Call` is a VarArg method where the first parameter is the method name and everything after that are the arguments: `object Call(string method, params object[] args)`.

- The first example is invalid because it will result in passing 3 `string` arguments instead of 1 `string[]` argument.
- The second one explains how to pass it as a single parameter. However, there's a shorter way to do it which is `(object)arr`. This does the same as `new object[] { arr }`.
- The third example is the same as the second one but with an int array instead of a string array so I think it's unnecessary.

This should be explained here. Also to avoid more confusion, as we have an example with a single array parameter, we should also have one with multiple parameters.

Also about these comments about silent fails without errors. Those won't result it compiler errors, but an error should be printed at runtime at the time of the call. But I'm not sure right now. Is that not happening?

Reply:
This is clarified a bit with the warning below, but I think it's better with comments next to each line.